src/test/librados/misc_cxx.cc

```diff
@@ -924,3 +926,37 @@ TEST_F(LibRadosMiscPP, Conf) {
   ASSERT_EQ(0, cluster.conf_get(option, actual));
   ASSERT_EQ(expected, actual);
 }
+
+// TEST_F(LibRadosMiscPP, ExecReadonlyWithWrite) {
+//   SKIP_IF_CRIMSON();
+//   bufferlist bl1, bl2;
+//   bl1.append("ceph");
+//   ObjectWriteOperation write;
+//   write.write(0, bl1);
+//   ASSERT_EQ(0, ioctx.operate("foo", &write));
+//
+//   // This should fail.
+//   ObjectReadOperation read1;
+//   cls_version_inc_op call;
+//   encode(call, bl2);
+//   read1.exec_ro("version", "inc", bl2);
+//   EXPECT_EQ(-EIO, ioctx.operate("foo", &read1, nullptr));
```

Comment:
I wonder whether to `EIO` is the best error code for a misbehaving client. `EINVAL`?